### PR TITLE
Update pyramid_tm to the latest best practices

### PR DESCRIPTION
**Pull request by bertjwregeer**
- Updates `tox.ini`
- Move to using `src` directory
- Add Github actions
- Remove Travis
- Remove appveyor
- Update .gitignore
- Add flake8
- Format code using black
- Update README
- Add missing term in docs
- Dropped Python 3.4 support
- Added Python 3.8

### CODE REVIEW

File: .github/workflows/ci-linux.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+              with:
+                  python-version: 2.7
+                  architecture: x64
+            - name: Setup python
+              uses: actions/setup-python@v1
+              with:
+                  python-version: 3.8
+                  architecture: x64
+            - run: pip install tox
+            - run: tox -e py38-pyramid110,py27,coverage
```
Comment (bertjwregeer) [inline .github/workflows/ci-linux.yml:46]: Note that this does not match the other CI files I've added to various other Pylons projects... cause we need `pyramid_retry` installed to validate full coverage.

File: docs/glossary.rst [hunk 2]
```diff
@@ -26,3 +26,11 @@ Glossary
       database management system.  In the context of the Pyramid documentation,
       "transaction" is also the name of a `Python package
       <https://pypi.org/project/transaction/>`__ used by ``pyramid_tm``.
+
+   dotted Python name
```
Comment (bertjwregeer) [inline docs/glossary.rst:30]: This term was missing and halting the build for docs.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+[build-system]
+requires = ["setuptools >= 41", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[tool.black]
+line-length = 79
```
Comment (bertjwregeer) [inline pyproject.toml:6]: I disagree with this...

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+skip-string-normalization = true
```
Comment (bertjwregeer) [inline pyproject.toml:7]: And this...

But I've done it make the blacken diff less large.

File: CHANGES.rst [hunk 5]
```diff
@@ -1,6 +1,13 @@
 Changes
 -------
 
+Unreleased
+^^^^^^^^^^
+
+- Drop Python 3.4 support
+
+- Add support for Python 3.7 and 3.8
```
Comment (mmerickel) [inline CHANGES.rst:9]: 3.7 was added in 2.2.1 below

File: tox.ini [hunk 6]
```diff
@@ -14,72 +18,87 @@ deps =
...
+commands =
+    make -C docs html BUILDDIR={envdir} SPHINXOPTS="-W -E"
+extras =
+    docs
+
+[testenv:run-flake8]
+skip_install = True
+basepython = python3.8
+commands =
+    flake8 src/colander/ tests
```
Comment (mmerickel) [inline tox.ini:74]: there's a few colander refs in this file atm

**Comment by mmerickel** [comment]
Ok I deleted the webhooks for appveyor and travis and configured appveyor to not run if appveyor.yml cannot be found.

**Pull request merged**